src/test/java/com/ipiecoles/java/java350/model/EmployeTest.java

```diff
@@ -77,4 +77,93 @@ public void testGetPrimeAnnuelleMatriculeNull(){
         //Then
         Assertions.assertThat(prime).isEqualTo(1000.0);
     }
+
+
+    //TEST DE LA METHODE augmenterSalaire
+
+    //Test augmentation basique en changeant le salaire et le pourcentage
+    //Test si salaire est null
+    //Test si pourcentage est 0
+    //Test si pourcentage est négative
+
+    @ParameterizedTest(name = "salaireBase {0}, pourcentageAugmentation {1}, salaireApresAugmentation {2}")
+    @CsvSource({
+            "1000.0, 10.0, 1100.0",
+            "1000.0, 50.0, 1500.0",
+            "2500.0, 10.0, 2750.0",
+            "2000.0, 15.5, 2310.0",
+            "1234, 28.5, 1585.69"
+    })
+    public void testAugmenterSalaireBase(double salaireBase, double pourcentage, double salaireApresAugmentation) {
+        //Given
+        Employe employe = new Employe();
+        employe.setSalaire(salaireBase);
+        //When
+        employe.augmenterSalaire(pourcentage);
+        //Then
+        Assertions.assertThat(employe.getSalaire()).isEqualTo(salaireApresAugmentation);
+    }
+
+    @Test
+    public void testAugmenterSalaireIfSalaireIsNull() {
+        //Given
+        Employe employe = new Employe();
+        employe.setSalaire(null);
+        //When
+        employe.augmenterSalaire(10.0);
+        //Then
+        Assertions.assertThat(employe.getSalaire()).isEqualTo(1673.34);
```

Comment:
On aurait pu mettre **_(Entreprise.SALAIRE_BASE * 10 / 100 ) + Entreprise.SALAIRE_BASE_** à la place de **1673.34** pour une meilleure compréhension du code.
```suggestion
        Double salairePlus10PC = Entreprise.SALAIRE_BASE * 10 / 100 + Entreprise.SALAIRE_BASE;
        Assertions.assertThat(employe.getSalaire()).isEqualTo(salairePlus10PC);
```

Reply:
Eventuellement... De manière générale il faut éviter le code "complexe" et les calculs dans les tests au profit de valeurs en dur justement.